corehq/apps/sso/tests/test_url_helpers.py

```diff
@@ -0,0 +1,46 @@
+from django.test import RequestFactory, TestCase
+
+from corehq.apps.sso.models import AuthenticatedEmailDomain
+from corehq.apps.sso.tests import generator
+from corehq.apps.sso.utils.url_helpers import add_username_hint_to_login_url
+
+
+class TestAddUsernameHintToLoginUrl(TestCase):
```

Comment:
Great tests, although I'm not sure they need to be `TestCase`s and involve setting up users, identity providers, and billing accounts. It seems a few of these tests could be written against `IdentityProvider.get_active_identity_provider_by_username` instead, while the tests here could mock out that function with the relevant values. For example, domain and identity provider mismatch seems to more be the responsibility of `get_active_identity_provider_by_username` than one for `add_username_hint_to_login_url`.

Reply:
yes! good point. I've addressed this, thanks!